agent/consul/leader_connect_ca_test.go

```diff
@@ -606,6 +607,88 @@ func TestCAManager_UpdateConfiguration_Vault_Primary(t *testing.T) {
 	require.Equal(t, connect.HexString(cert.SubjectKeyId), newRoot.SigningKeyID)
 }
 
+func TestCAManager_Initialize_Vault_WithIntermediateAsPrimaryCA(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+	ca.SkipIfVaultNotPresent(t)
+
+	vault := ca.NewTestVaultServer(t)
+	vclient := vault.Client()
+	generateExternalRootCA(t, vclient)
+
+	meshRootPath := "pki-root"
+	primaryCert := setupPrimaryCA(t, vclient, meshRootPath)
+
+	_, s1 := testServerWithConfig(t, func(c *Config) {
+		c.CAConfig = &structs.CAConfiguration{
+			Provider: "vault",
+			Config: map[string]interface{}{
+				"Address":             vault.Addr,
+				"Token":               vault.RootToken,
+				"RootPKIPath":         meshRootPath,
+				"IntermediatePKIPath": "pki-intermediate/",
+				// TODO: there are failures to init the CA system if these are not set
+				// to the values of the already initialized CA.
```

Comment:
```suggestion
				// to the values of the CA initialized by setupPrimaryCA.
```

Reply:
I think these suggested edits are the opposite of what is happening. 

This test case initializes the CA in Vault itself, so we're setting the config values to those of the CA in Vault. `setupPrimaryCA` will ignore the values because Vault already has a CA setup.

I opened #12246 yesterday to track this TODO. Once that work is done I can come back to these tests and remove the TODO and the two lines below.

Reply:
Ah got it, thanks for clarifying!